src/yaml/aslYamlLanguageService.ts

```diff
@@ -0,0 +1,228 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import * as prettier from 'prettier'
+import {
+    DocumentLanguageSettings,
+    getLanguageService as getLanguageServiceVscode,
+    JSONDocument,
+    JSONSchema,
+    LanguageService,
+    LanguageServiceParams,
+} from 'vscode-json-languageservice'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+    CompletionList,
+    DocumentSymbol,
+    FormattingOptions,
+    Hover,
+    Position,
+    Range,
+    SymbolInformation,
+    TextEdit
+} from 'vscode-languageserver-types'
+import {
+    parse as parseYAML,
+    SingleYAMLDocument,
+    YAMLDocument,
+} from 'yaml-language-server/out/server/src/languageservice/parser/yamlParser07'
+import { YAMLCompletion } from 'yaml-language-server/out/server/src/languageservice/services/yamlCompletion'
+import { YAMLSchemaService } from 'yaml-language-server/out/server/src/languageservice/services/yamlSchemaService'
+import { matchOffsetToDocument } from 'yaml-language-server/out/server/src/languageservice/utils/arrUtils'
+import doCompleteAsl from '../completion/completeAsl'
+
+export const getLanguageService = function(params: LanguageServiceParams, schema: JSONSchema, aslLanguageService: LanguageService): LanguageService {
+    const builtInParams = {}
+
+    const languageService = getLanguageServiceVscode({
+        ...params,
+        ...builtInParams,
+    })
+
+    const requestServiceMock = async function(uri: string): Promise<string> {
+        return new Promise<string>(c => {
+            c(JSON.stringify(schema))
+        })
+    }
+    const schemaService = new YAMLSchemaService(requestServiceMock, params.workspaceContext)
+    // initialize schema
+    schemaService.registerExternalSchema('yasl', ['*.asl.yaml', '*.asl.yml'], schema)
+    schemaService.getOrAddSchemaHandle('yasl', schema)
+
+    const completer = new YAMLCompletion(schemaService)
+
+    languageService.doValidation = async function(
+        textDocument: TextDocument,
+        jsonDocument: JSONDocument,
+        documentSettings: DocumentLanguageSettings
+    ) {
+        const yamlDocument: YAMLDocument = parseYAML(textDocument.getText())
+        const validationResult: any[] = []
+
+        for (const currentYAMLDoc of yamlDocument.documents) {
+            const validation = await aslLanguageService.doValidation(textDocument, currentYAMLDoc)
+            const syd = ((currentYAMLDoc as unknown) as any) as SingleYAMLDocument
```

Comment:
Also out of curiosity, how come we have to do so many type casts before casting to SYD?

Reply:
This only seems to need `(currentYAMLDoc as unknown) as SingleYAMLDocument;` as shown [here](https://github.com/redhat-developer/yaml-language-server/blob/441f697f2587017283437cd293d4fc213be5839a/src/languageservice/services/yamlValidation.ts#L69). Changing this to match the YAML LS.